workflow/artifacts/common/load_to_stream.go

```diff
@@ -0,0 +1,36 @@
+package common
+
+import (
+	"io"
+	"os"
+
+	"k8s.io/apimachinery/pkg/util/rand"
+
+	wfv1 "github.com/argoproj/argo-workflows/v3/pkg/apis/workflow/v1alpha1"
+)
+
+// wrapper around os.File enables us to remove the file when it gets closed
+type selfDestructingFile struct {
+	os.File
+}
+
+func (w selfDestructingFile) Close() error {
+	err := w.File.Close()
+	_ = os.Remove(w.Name())
+	return err
+}
+
+// Use ArtifactDriver.Load() to get a stream, which we can use for all implementations of ArtifactDriver.OpenStream()
+// that aren't yet implemented the "right way" and/or for those that don't have a natural way of streaming
+func LoadToStream(a *wfv1.Artifact, g ArtifactDriver) (io.ReadCloser, error) {
+	filename := "/tmp/" + rand.String(32)
```

Comment:
Can we add a log line here? I'd like it to say:

"Efficient artifact streaming is not supported for %s: see INSERT GITHUB ISSUE HERE", g.GetType()

Maybe users will see it and implement for us?

Reply:
Info or Warn level?

Reply:
Are you referring to the current "Artifact Streaming" github issue or some new issue we leave outstanding related to implementing the rest of these?

Reply:
Warn level I think is probably correct. Yes, link to the artifact streaming issue.

Reply:
added a new issue and using Info level